FACTOR.py

```diff
@@ -328,7 +328,7 @@ def tx_compute_merkle_root(tx_hashes):
 
         for i in range(len(tx_hashes) // 2):
             # Concatenate the next two
-            concat = tx_hashes.pop(0) + tx_hashes.pop(0)
+            concat = tx_hashes[i] + tx_hashes[i + 1]
```

Comment:
You want ``[2*i]`` and ``[2*i+1]`` for the indexes. But otherwise, good find. I am testing now.

Reply:
hi, using [2*i] and [2*i+1] gives me the exact same hash...

Reply:
Yes!! Because you only have two hashes, or one in the case of coinbase, but if you had 4 transactions then you would see the difference. :)

Reply:
I tested your first proposal with my suggestion about the indexes, that fails. I can test the endianess as you suggest, though in my previous testing it did not work either, but to confirm let me test again as you suggest.

Reply:
That is not it either.

Reply:
Hi,
I'm seeing a problem with the coinbase, I think is better not to have a witness hash in it:

In the readme: 
hex: 0014a8a44c20e7b12de5405bb864cd9d5be3c5bd055a
asm: 0 a8a44c20e7b12de5405bb864cd9d5be3c5bd055a

The correct will be: 
hex: 76a9a8a44c20e7b12de5405bb864cd9d5be3c5bd055a88ac
asm:
OP_DUP
OP_HASH160
a8a44c20e7b12de5405bb864cd9d5be3c5bd055a
OP_EQUALVERIFY
OP_CHECKSIG

Reply:
Another thing im seeing is that the version of the transaction of the coinbase is 2 and i think everbody is using 1, I do not know if that affects in anything.

Reply:
I think the problem that we are having is this, a bad mix of segwit from the side of FACT0RN daemon with non segwit old code in the miner code.

https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki#commitment-structure

In the Readme the address specified is a witness address but the block creation in the miner does not include any merkle tree for witness information.